andela/ah-alpha
# #162948912 user profile should be created automatically when a…

The PR is hosted here: https://ah-alpha-pr-15.herokuapp.com/swagger/
#### What does this PR do?

Ensures a user profile is created when a user registers to the site

#### Description of Task to be completed?

* User profile should be created when a user registers
* User should be able to edit their profile
* User should be able to see other peoples profile but cannot edit
#### How should this be manually tested?

* `git fetch origin ft-user-profile-162948912`
* Run `pip install -r requirements.txt` to update requirements.txt 
* set up postgres database on the psql command line
* Create a virtual environment and create a .env file and add the database credentials
* export database credentials. please note that there are no spaces before and after =
``` export DB_NAME=" name of your database"
      export DB_PASSWORD="passord to your database user"
      export DB_HOST="your local server eg (127.0.0.1)"
      export DB_USER="your database username eg (postgres)"
```
* Note- if you have no password set for your database leave the password field blank example: 
  `export DB_PASSWORD=""`

* In the project's root directory run `python manage.py runserver`
* Create a user by sending a POST request to the url `http://127.0.0.1:8000/api/v1/users/ `with the following in the body.
```
{
    "user":{
        "username":"erastus",
        "email":"example@email.com",
        "password":"Password"
    }
}

```
## NB: you must be LOGGED IN to access the profiles endpoints

* - To view the user profiles availabe visit this url
   - `http://127.0.0.1:8000/api/v1/profiles/` .
   - This is a GET request
   - example response data:
```
{
    "profile": [
        {
            "username": "erastus",
            "image": "http://127.0.0.1:8000/api/v1/profiles/default.jpg",
            "bio": "",
            "location": "",
            "company": "",
            "created_at": "2019-01-22T07:21:07.742652Z",
            "updated_at": "2019-01-22T07:21:07.742674Z"
        },
        {
            "username": "admin",
            "image": "http://127.0.0.1:8000/api/v1/profiles/default.jpg",
            "bio": "",
            "location": "",
            "company": "",
            "created_at": "2019-01-22T07:35:18.957071Z",
            "updated_at": "2019-01-22T07:35:18.957092Z"
        }
    ]
}

```
- To view a single users profile
- Navigate to `http://127.0.0.1:8000/api/v1/profiles/<username>/`
- This is a  GET request
- Example response message
```
{
    "profile": {
        "username": "erastus",
        "image": "http://127.0.0.1:8000/api/v1/profiles/erastus/default.jpg",
        "bio": "",
        "location": "",
        "company": "",
        "created_at": "2019-01-22T07:21:07.742652Z",
        "updated_at": "2019-01-22T07:21:07.742674Z"
    }
}
```

- To update your  profile
- Navigate to `http://127.0.0.1:8000/api/v1/profiles/edit/<username>/`
- This is a PATCH request
- Example of data format

```
{
        'profile':
        {
            "bio": "i am a tech enthusiast",
            "image": "default.jpg",
            "company": "Andela",
            "location": "cape Town"
        }
 }
```
#### What are the relevant pivotal tracker stories?
[#162948912 ](https://www.pivotaltracker.com/story/show/162948912)
